lib/net.c

```diff
@@ -1038,10 +1047,22 @@ gatekeeper_get_rss_config(uint16_t port_id,
 	return ret;
 }
 
-/*
- * @port_idx represents index of port in iface->ports when it is >= 0.
- * When it is -1, @port_id is a bonded port and has no entry in iface->ports.
- */
+int
+gatekeeper_setup_user_perms(struct net_config *net_conf,
+	const char *user, unsigned int mode)
+{
+	if (user == NULL)
+		return -1;
+
+	net_conf->user = rte_strdup(__func__, user);
+	if (net_conf->user == NULL)
```

Comment:
1. We should test that the user exists calling `getpwnam()`.

2. Instead of saving the name of the user, we can save its `pw_uid` and `pw_gid`. This makes the code a bit simpler.

Reply:
We need to call ``initgroups()`` function, which needs the ``user`` as a parameter.

Reply:
I have moved the ``initgroups()`` function to here without storing the ``user`` name.